### ISSUE TITLE
No summary was found for this file

### ISSUE THREAD

**Issue by DaveWP** [phpDocumentor/phpDocumentor#1525]
I have searched and it appears that phpdocumentor requires a PHP statement to be present at the top of a file.  If not, then this error will be generated.

For example, the following code works:

```
<?php
/**
 * The Header for our theme.
 * 
 * Displays all of the &lt;head&gt; section and everything up til &lt;div id="main"&gt;
 *
 * @package theme
 * @subpackage header block
 * @since 1.0
 *  
 */
if ($a==""){echo "rubbish";}

?><!DOCTYPE html>
<html <?php language_attributes(); ?>>
```

Whilst the attached header without the line if($a==""){echo "rubbish";) statement causes this error to be generated.

```
<?php
/**
 * The Header for our theme.
 * 
 * Displays all of the &lt;head&gt; section and everything up til &lt;div id="main"&gt;
 *
 * @package theme
 * @subpackage header block
 * @since 1.0
 *  
 */

?><!DOCTYPE html>
<html <?php language_attributes(); ?>>
```

**Comment by niallscc** [comment]
Hey you probably already figured this out, however in case anyone else finds this page here is the answer that worked for me and that I found on stackoverflow: 

But does the top of your file have two such docblocks? The first docblock is expected to be the one for the file itself, and a second docblock should pair up with the first documentable code element that appears after it.

However, if you only have one docblock at the top of the file, it will get paired up with the first code element found, thus the "file itself" will seem to be missing its docblock. That is what that error is supposed to indicate.

**Comment by MCMainiac** [comment]
I have a similar problem:

    <?php
    /**
     * This file contains the LoginLib class
     * 
     * Longer description
     */
    
    /**
     * Short description of class
     * 
     * Longer description of class...
     */
    class LoginLib {

I get absolutely NO documentation and the only error was the one mentioned above, which also lead me to this issue.

I'm not sure if it is same or just a similar error. I also found the stackoverflow question mentioned by @niallscc, but it didn't help me since I have to docblocks...

**Comment by stephanepiriou** [comment]
This rule doesn't make sense.

I have a php file (basically the header of a view in a custom MVC project) which contain only html code, `header.php`:
```html
<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="fr" lang="fr">
<head>
    <meta charset="UTF-8">
```
phpdoc tells me there is no summary to the file and record an error.


If I add php tag to document with a docBlock inside with the name of the package :
```php
<?php
/**
 * Header of the php view
 * @package App\View\Common
 */
?>
<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="fr" lang="fr">
<head>
    <meta charset="UTF-8">
```
It still raise the same error.


But if I add a dummy function and document it:
```php
<?php
/**
 * Header of the php view
 * @package App\View\Common
 */
/**
 * Useless function
 * @return void
 */
function f(){}
?>
<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="fr" lang="fr">
<head>
    <meta charset="UTF-8">
```
then the error disappear.

In short, one can make a phpdoc summary of a php page only if the page have another docBlock about anything else after. 

The is something Kafkaesque about this rule.

**Comment by jaapio** [comment]
Remember that phpDocumentor is an api doc generator, a php file without any php in it doesn't have an api. So it should not be part of your api docs. It's a bit strange to me that everyone expects the file to be in the docs. 
The validation here is obviously wrong, since the file doesn't contain any php at all it should not be validated. But since you are adding some php, only a licence block the file is validated. 

A file cannot have comments in the AST of php, comments belong to an element, like a function, class, or any other structural element. This is why phpdocumentor is checking for multiple and expects the first comment to be the file header. But actually, there is no such thing as file header.

So the validation should be changed to cover this case. Feel free to create a PR. The phpDocumentor team is focusing on phpDoc v3 with will not have validation in the first release. Because we think there are better tools atm to do validation of docblocks. We will re-add it in the future. but for now it doesn't make sense to spend much time to add a feature like this.

**Comment by jaapio** [comment]
This bug was reported on phpDocumentor v2, and will not be addressed. If this is still applicable on the latest version. Please feel free to open a new issue.
